js/navigation.js

```diff
@@ -1,77 +1,207 @@
 namespacer('bcpl');
 
-bcpl.navigation = (($) => {
-	const navButtonSelector = '.nav-and-search nav button';
-	const navButtonAndListSelector = '.nav-and-search nav li.active button, .nav-and-search nav li.active ul';
-
-	const keyCodes = {
-		enter: 13
+bcpl.navigation = (($, keyCodes) => {
+	const navButtonSelector = '#responsive-sliding-navigation button';
+	const closestMenuNodeSelector = '#responsive-sliding-navigation>ul>li';
+	const searchArtifactsSelector = '#activate-search-button, #search-box';
+	const heroCalloutContainerSelector = '.hero-callout-container';
+	const activeLinksSelector = '.active, .clicked';
+	const activeMenuButtonSelector = 'li.active button';
+	const mobileWidthThreshold = 768;
+
+	const isMobileWidth = () => parseFloat($('body').width()) <= mobileWidthThreshold;
+
+	const isSlideNavigationVisible = () => $('body').hasClass('nav-visible');
+
+	const focusFirstActiveMenuLink = () => $('#responsive-sliding-navigation li.active a').first().focus();
+
+
+	const findClosestButtonToLink = ($link) => $link.closest(closestMenuNodeSelector).find('button');
+
+	const activateSubmenu = ($button, afterAnimationCallback) => {
+		const animationOptions = isSlideNavigationVisible() ? { right: '0px' } : {};
+		const animationSpeed = isSlideNavigationVisible() ? 250 : 0;
+		$button
+			.attr('aria-expanded', true)
+			.closest('li')
+			.addClass('active')
+			.find('.submenu-wrapper')
+			.animate(animationOptions, animationSpeed, function afterAnimation() {
```

Comment:
should this be it's own function?

Reply:
In this case, I didn't extract it since I have to pass the callback in. But, it's the difference of

`foo.bar(callback);`
vs
`foo.bar(() => { otherStuff(); callback(); });`

I'll extract it.

Reply:
Done.